pkg/controller/container-runtime-config/container_runtime_config_controller.go

```diff
@@ -274,11 +275,20 @@ func (ctrl *Controller) cascadeDelete(cfg *mcfgv1.ContainerRuntimeConfig) error
 	if len(cfg.GetFinalizers()) == 0 {
 		return nil
 	}
-	mcName := cfg.GetFinalizers()[0]
-	err := ctrl.client.MachineconfigurationV1().MachineConfigs().Delete(context.TODO(), mcName, metav1.DeleteOptions{})
-	if err != nil && !errors.IsNotFound(err) {
+	finalizerName := cfg.GetFinalizers()[0]
+	mcs, err := ctrl.client.MachineconfigurationV1().MachineConfigs().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
 		return err
 	}
+	for _, mc := range mcs.Items {
+		if string(mc.ObjectMeta.GetUID()) == finalizerName || mc.GetName() == finalizerName {
```

Comment:
I think I'm missing someting 🤔 how's the finalizerName ever equal to a MC name ? I guess why this routine got changed

Reply:
an MC name is also unique and we're picking only `[0]` which should be just one MC anyway

Reply:
The list of Finalizers hold the MC name, and since we only create one MC which we overwrite, the list of Finalizers will have the same name so we can just pick `[0]` and use that to delete the MC. This is for the MC snippet, e.g 99-worker-blahblah-containerruntime